extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java

```diff
@@ -520,13 +520,8 @@ private String generateImplementation(ClassInfo defaultBundleInterface, String d
         String bundleName = nameValue != null ? nameValue.asString() : MessageBundle.DEFAULT_NAME;
         AnnotationValue defaultKeyValue = bundleAnnotation.value(BUNDLE_DEFAULT_KEY);
 
-        String baseName;
-        if (bundleInterface.enclosingClass() != null) {
```

Comment:
@mkouba I found a few places like this, name() already has the enclosing name (`EnclosingName$InnerName`) so this was ending up as `EnclosingName_EnclosingName$InnerName`, which then confused the application class predicate.

Reply:
I'm pretty sure we used this for a reason so either something changed in the Jandex behavior or there's a corner case we tried to solve. Let me take a look...

Reply:
Ok, so I think that the correct version should be:
```java
if (bundleInterface.enclosingClass() != null) {
   baseName = DotNames.simpleName(bundleInterface.enclosingClass()) + "_"
          + DotNames.simpleName(bundleInterface);
} else {
   baseName = DotNames.simpleName(bundleInterface);
}
```
`DotNames.simpleName(bundleInterface.name())` -> `DotNames.simpleName(bundleInterface)`

And the resulting name should be `EnclosingName_InnerName` (at least that was the intent and we use this convention in ArC and some other extensions). The `DotNames.simpleName(DotName)` is not entirely correct because the dollar sign is a valid character for class names so we cannot detect a nested class here. 

> which then confused the application class predicate.

Do you remember the details? I'd like to verify this...

Reply:
I will address this in a separate PR. Pls revert this change.

Reply:
FTR https://github.com/quarkusio/quarkus/pull/16583

Reply:
Does that actually work with the application class predicate though?

I am thinking now that I made a big mistake where I went for a flatish classpath in tests and the multiple ClassLoader's in dev mode, as things that work fine in tests and prod would actually fail in dev mode. 

Also this is part of a fairly massive changeset that bypasses all the incremental build stuff as it touches everything and really needs to be in by the alpha, so I am going to leave it as is and if this ever manages to pass CI and get in your change can go on top of it :-)

Reply:
> Does that actually work with the application class predicate though?

It should work with the predicate used in the `MessageBundleProcessor`.

> Also this is part of a fairly massive changeset that bypasses all the incremental build stuff as it touches everything and really needs to be in by the alpha, so I am going to leave it as is and if this ever manages to pass CI and get in your change can go on top of it :-)

Well, I'd like to merge that PR and then there will be likely conflicts... so we'll see how it goes ;-).